bin/cylc-get-host-metrics

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2018 NIWA
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""cylc get-host-metrics [OPTIONS]
+
+   Get metrics for localhost, in the form of a JSON structure with
+   top-level keys as requested via the OPTIONS:
+
+   1. --load
+          1, 5 and 15 minute load averages (as keys) from the 'uptime' command.
+   2. --memory
+          Total free RAM memory, in kilobytes, from the 'free -k' command.
+   3. --disk-space=PATH / --disk-space=PATH1,PATH2,PATH3 (etc)
+          Available disk space from the 'df -Pk' command, in kilobytes, for one
+          or more valid mount directory PATHs (as listed under 'Mounted on')
+          within the filesystem of localhost. Multiple PATH options can be
+          specified via a comma-delimited list, each becoming a key under the
+          top-level disk space key.
+
+   If no options are specified, --load and --memory are invoked by default."""
+
+
+import sys
+if '--use-ssh' in sys.argv[1:]:
+    sys.argv.remove('--use-ssh')
+    from cylc.remote import remrun
+    if remrun():
+        sys.exit(0)
+
+import os
+import re
+from subprocess import Popen, PIPE, CalledProcessError
+import json
+
+import cylc.flags
+from cylc.option_parsers import CylcOptionParser as COP
+
+
+"""Templates for extracting desired metric data via regular expression.
+
+   Example command outputs, with '*' enclosing data to extract with templates:
+
+   LOAD_TEMPLATES
+     $ uptime
+     17:52:58 up  8:08,  2 users,  load average: *0.13*, *0.09*, *0.07*
+
+   MEMORY_TEMPLATES
+     $ free -k
+                  total       used       free     shared    buffers     cached
+     Mem:       6116852    5017920    1098932    54964     130328    1837404
+     -/+ buffers/cache:    3050188  *3066664*
+     Swap:      4190204      14624    4175580
+
+   DISK_TEMPLATES, e.g. for mount directory path option '--disk-space=/boot'
+     $ df -Pk /boot
+     Filesystem            1024-blocks      Used Available Capacity Mounted on
+     /dev/md0                    96978     33904   *57856*      37% /boot."""
+LOAD_TEMPLATES = ['uptime', (1, 5, 15), re.compile(
+                  r'load average:\s(\d+\.?\d+),\s(\d+\.?\d+),\s(\d+\.?\d+)')]
```

Comment:
Looking back at the regex, I think you are trying to make the decimal point optional `\d+\.?\d+`, as `\.?` is followed by `\d+` it requires a second digit to match. Probably not an issue (would uptime ever display integers?) but easily fixed e.g. `[\d\.]+`

Reply:
Well spotted, I'll look at the docs and see if the decimal place is always included.